tests/discovery_test.js

```diff
@@ -4,13 +4,341 @@ import chai, { expect } from "chai";
-    expect(hash1).to.eql(getUserIDHash('shweta'));
+const root = typeof window === "object" ? window : global;
+var localStorage;
+var LocalStorage = require('node-localstorage').LocalStorage;
+localStorage = new LocalStorage('./lib/localStorage');
+
+
+describe("discovery library", () => {
+  let sandbox;
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+    localStorage.clear();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe("getUserIDHash", () => {
+    it("should return the same hash for the same userid", () => {
+      const hash = getUserIDHash("shweta");
+      expect(hash).to.eql(getUserIDHash("shweta"));
+    });
+
+    it("should return different hashes for different values", () => {
+      const hash = getUserIDHash("shweta");
+      expect(hash).to.not.eql(getUserIDHash("alexis"));
+    });
+
+    it("should return an uuid", () => {
+      const hash = getUserIDHash("shweta");
+      expect(hash).to.match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/);
+    });
+  });
+
+  describe("registerUserURL", () => {
+
+    const centralRepositoryURL = "http://central.kinto-storage.com/v1";
+    const headers = {'Authorization': 'Bearer 1234567'};
+    let userStorageURL;
+
+    describe("With an already existing user URL", () => {
+      userStorageURL = "https://my-kinto-instance.com/v1";
+      beforeEach(() => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(200, {
+          data: {url: userStorageURL}
+        }, {}));
+      });
+
+      it("should return the existing URL", () => {
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.become(userStorageURL);
+      });
+    });
+
+    describe("when server returns a 5xx", () => {
+      userStorageURL = "https://my-kinto-instance.com/v1";
+      const defaultServer = "https://default-kinto-instance.com/v1"
+
+
+      it("should return an error in case of 501 ", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(501, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+
+      it("should return an error in case of 500", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(500, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+
+      it("should return an error in case of 503", () => {
+        sandbox.stub(root, "fetch").returns(fakeServerResponse(503, {
+          data: {url: userStorageURL}
+          }, {} ));
+        return registerUserURL("userID", centralRepositoryURL, headers, userStorageURL)
+        .should.be.rejectedWith(Error);
+      })
+     });
+
+
+         describe("does not fit in any case", () => {
```

Comment:
The list of possible errors is well defined, so we should test for 401/403/500/503 with the error message set accordingly.